src/jaegertracing/TracerFactoryTest.cpp

```diff
@@ -65,19 +67,70 @@ TEST(TracerFactory, testValidConfig)
         "refreshInterval": 60
     }
   })";
-    TracerFactory tracerFactory;
+    const auto tracerFactory = new (std::nothrow) TracerFactory(true);
     std::string errorMessage;
-    auto tracerMaybe = tracerFactory.MakeTracer(config, errorMessage);
+    auto tracerMaybe = tracerFactory->MakeTracer(config, errorMessage);
     ASSERT_EQ(errorMessage, "");
     ASSERT_TRUE(tracerMaybe);
 }
+
+TEST(TracerFactory, testWithoutReadFromEnv)
+{
+    const char* config = "";
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "AService");
+    const auto tracerFactory = new (std::nothrow) TracerFactory();
+    std::string errorMessage;
+    auto tracerMaybe = tracerFactory->MakeTracer(config, errorMessage);
+    ASSERT_FALSE(tracerMaybe);
+    ASSERT_NE(errorMessage, "");
+
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "");
+}
+
+TEST(TracerFactory, testWithReadFromEnv)
+{
+    const char* config = "";
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "AService");
+    const auto tracerFactory = new (std::nothrow) TracerFactory(true);
```

Comment:
There is a memory leak here. There is no need to call new. TracerFactory tracerFactory(new) is sufficient.